rest/src/main/java/org/opentestsystem/authoring/testspecbank/service/impl/TestSpecificationServiceImpl.java

```diff
@@ -506,8 +506,13 @@ private void populateTestSpecification(final TestSpecification testSpecification
             } catch (final IOException e) {
                 throw new LocalizedException("testspec.spec.xml.notfound", new String[] { testSpecification.getSpecificationXmlGridFsId() }, e);
             }
-            final byte[] decompressedXml = decompress(ret.toByteArray());
-            testSpecification.setSpecificationXml(decompressedXml);
+            if (ret.size() > 0) {
+                final byte[] decompressedXml = decompress(ret.toByteArray());
+                testSpecification.setSpecificationXml(decompressedXml);
+            } else {
+                LOGGER.error("Test specification xml contains zero bytes");
```

Comment:
Is there more data that we can log with this error?  For example, it would be very helpful to be able to track from this error log the test specification that is in error.  Basically, allow someone that sees the error to quickly find the data in question.

Reply:
Agreed.  I'll check if I can add the assessment identifying info.